File: api/models/capitulo.go
package models

import "time"

type Capitulo struct {
	ID             uint      `gorm:"primaryKey" json:"id"`
	MangaID        uint      `gorm:"not null" json:"manga_id"`
	Titulo         string    `gorm:"type:varchar(45);not null" json:"titulo"`
	Numero         uint32    `gorm:"primaryKey;autoIncrement:false;" json:"numero"`
	Url            string    `gorm:"type:varchar(120);not null" json:"url"`
	DataPublicacao time.Time `gorm:"type:datetime;not null" json:"data_publicacao,omitempty"`
}

File: app/lib/models/capitulo.dart
class Capitulo {
  final String titulo;
  final int numero;
  final String url;
  final DateTime dataPublicacao;

  Capitulo(this.titulo, this.numero, this.url, this.dataPublicacao);

  Capitulo.fromJson(Map<String, dynamic> json)
      : titulo = json['titulo'],
        numero = json['numero'],
        url = json['url'],
        dataPublicacao = DateTime.parse(json['data_publicacao']);
}
